pacifica/metadata/rest/doi_queries/doi_registration_update.py

```diff
@@ -0,0 +1,114 @@
+            DOIRegistrationUpdate._extract_authors(creators_block, doi_string)
+            DOIRegistrationUpdate.change_doi_entry_info(
+                doi_string, doi_info, current_status, release_status)
+        else:
+            raise HTTPError(404, 'DOI Entry %s does not exist' % doi_string)
+
+        return doi_string
+
+    @staticmethod
+    def _extract_doi_info_from_xml(record_object):
+        doi_info = {}
+        children = (child for child in record_object if child.text)
+        creators_block = record_object.find('creatorsblock')
+        record_object.remove(creators_block)
+        for child in children:
+            if 'date' in child.tag:
+                info = parse(child.text).strftime('%Y-%m-%d')
+            else:
+                info = child.text
+            doi_info[child.tag] = info
+        return doi_info, creators_block
+
+    @staticmethod
+    def _check_for_doi_entry(doi_string):
+        check_query = DOIEntries.select().where(DOIEntries.doi == doi_string)
+        return check_query.count() > 0
+
+    @staticmethod
+    def _extract_authors(creatorsblock_element, doi_string):
+        author_list = [{x.tag: x.text for x in el}
+                       for el in creatorsblock_element]
+        DOIRegistrationUpdate._update_author_info(author_list, doi_string)
+
+    @staticmethod
+    def _update_author_info(author_list, doi_string):
+        # cross check author list with doi_authors
+        # adding new and retrieving existing
+        author_info_mapping = {
+            'first_name': 'first_name',
+            'last_name': 'last_name',
+            'affiliation_name': 'affiliation',
+            'private_email': 'email',
+            'orcid_id': 'orcid'
+        }
+        author_id_list = []
+        for author_info in author_list:
+            my_author_info = {}
+            for key in author_info:
+                my_key = author_info_mapping[key]
+                my_author_info[my_key] = author_info[key]
+            author, _created = DOIAuthors.get_or_create(**my_author_info)
+            author_id_list.append(author.id)
+
+        # add doi to author mappings
+        author_order = 0
+        for author_id in author_id_list:
+            author_order += 1
+            author_map_insert_item = {'author_order': author_order}
+            author_map_lookup_item = {'author': author_id, 'doi': doi_string}
+            DOIAuthorMapping.get_or_create(
+                defaults=author_map_insert_item, **author_map_lookup_item)
+
+    # CherryPy requires these named methods.
+    # pylint: disable=invalid-name
+    @staticmethod
+    @tools.json_out()
+    def POST():
+        """Update existing DOI Entries."""
+        valid_content_types = ['application/xml', 'text/xml']
+        headers = request.headers
+        if headers['Content-type'] not in valid_content_types:
+            raise HTTPError(415, 'Expected an entity of content type %s' %
```

Comment:
Just a comment for future note. Seems CherryPy has some built in XML support but that effort is around XML-RPC specifically and they don't provide clean tools similar to their JSON in and out methods.

Reply:
I looked for a built-in xml handler like the one for json. That would have been nice to have. The RPC stuff was pretty unwieldy.